File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const uniqueNums = new Set(nums);
        let maxLength = 0;
        nums.forEach(num => {
            if(!uniqueNums.has(num - 1)){
                let length = 1;
                while(uniqueNums.has(num + length)) {
                    length++;
                }
                maxLength = Math.max(length, maxLength);
            }
        })
        return maxLength;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        total_numbers = len(nums)
        dp = [ 1 for _ in range(total_numbers)]
        for i in range(1, total_numbers):
            for j in range(i):
                if nums[j] < nums[i] and dp[j] + 1 > dp[i]:
                    dp[i] = dp[j] + 1 
        return max(dp)